3-UsingAPIs/Week2/homework/ex3-rollAnAce.js

```diff
@@ -10,23 +10,25 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
 3. Refactor the function `main()` to use async/await and try/catch.
 ------------------------------------------------------------------------------*/
 // ! Do not change or remove the next two lines
+
 const rollDie = require('../../helpers/pokerDiceRoller');
 
-function rollDieUntil(wantedValue) {
-  // TODO: rewrite this function using async/await
-  return rollDie().then((value) => {
-    if (value !== wantedValue) {
-      return rollDieUntil(wantedValue);
-    }
-    return value;
-  });
+async function rollDieUntil(wantedValue) {
+  let value = await rollDie();
+
+  while (value !== wantedValue) {
+    value = await rollDie();
+  }
```

Comment:
Excellent!

This will also work:

```suggestion
  let value = null;

  while (value !== wantedValue) {
    value = await rollDie();
  }
```